src/chainstate/coordinator/mod.rs

```diff
@@ -0,0 +1,307 @@
+use std::collections::VecDeque;
+use std::thread;
+use std::sync::{Arc, Mutex};
+use std::sync::mpsc::{self, Sender};
+use std::process;
+
+use burnchains::BurnchainHeaderHash;
+use chainstate::burn::{BlockHeaderHash, BlockSnapshot};
+use chainstate::stacks::{StacksBlock, TransactionPayload};
+use chainstate::stacks::db::StacksHeaderInfo;
+use chainstate::stacks::events::{StacksTransactionReceipt};
+
+#[derive(Debug, Clone)]
+struct PoxId;
+
+#[derive(Debug, Clone)]
+pub struct SortitionId (
+    BurnchainHeaderHash,
+    PoxId
+);
+
+#[derive(Debug, Clone)]
+pub struct StacksBlockId {
```

Comment:
This is already defined as `StacksBlockId([u8; 32])`, and will be calculated as the hash of the consensus hash for the burnchain block (which itself is derived from the PoX anchor block vector) and the Stacks block hash.